### PR TITLE
crimson/osd: submit_transaction() refactoring

### PR DESCRIPTION
**Pull request by Matan-B**
By moving `log_entries` to `OpsExecuter` we will be able to separate the pg log entry creation from `submit_transaction()`.
Moreover, when introducing snapshots to Crimson a `CLONE` log entry will make use of the OpsExecuter's `log_entries`.

Signed-off-by: Matan Breizman <mbreizma@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by athanatos** [comment]
It seems like the contents of prepare_transaction here could be in flush_changes_n_do_ops_effects.  log_entries doesn't need to be a member of OpsExecutor, it could be generated within flush_changes_n_do_ops_effects (or in a helper invoked from there) and immediately passed to MutFunc and then into submit_transaction.

**Comment by Matan-B** [comment]
jenkins test make check

**Comment by Matan-B** [comment]
jenkins test windows

**Comment by Matan-B** [comment]
> It seems like the contents of prepare_transaction here could be in flush_changes_n_do_ops_effects. log_entries doesn't need to be a member of OpsExecutor, it could be generated within flush_changes_n_do_ops_effects (or in a helper invoked from there) and immediately passed to MutFunc and then into submit_transaction.

Fixed. Made sense to also move `fill_op_params_bump_pg_version()` out of pg.

File: src/crimson/osd/ops_executer.h [hunk 1]
```diff
@@ -334,12 +340,13 @@ OpsExecuter::flush_changes_n_do_ops_effects(MutFunc&& mut_func) &&
 	seastar::now(),
 	interruptor::make_interruptible(osd_op_errorator::now()));
   if (want_mutate) {
-    osd_op_params->req_id = msg->get_reqid();
-    osd_op_params->mtime = msg->get_mtime();
+    std::vector<pg_log_entry_t> log_entries;
+    fill_op_params_bump_pg_version();
+    prepare_transaction(log_entries, ops);
```
Comment (athanatos) [inline src/crimson/osd/ops_executer.h:345]: Why not just return log_entries?

**Review by athanatos — APPROVED** [review]
LGTM other than than returning log_entries from prepare_transaction.

**Comment by Matan-B** [comment]
jenkins retest this please

**Comment by Matan-B** [comment]
jenkins test crimson perf

**Comment by Matan-B** [comment]
jenkins test api

**Comment by Matan-B** [comment]
jenkins test classic perf

**Comment by Matan-B** [comment]
jenkins test crimson perf

**Pull request merged**